Mathlib/RingTheory/PowerSeries/GaussNormC.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2025 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero
+-/
+import Mathlib.Analysis.Normed.Ring.Basic
+import Mathlib.RingTheory.PowerSeries.Order
+/-!
+# Gauss norm for power series
+This file defines the Gauss norm for power series. Given a power series `f` in `R⟦X⟧`, a function
+`v : R → ℝ` and a real number `c`, the Gauss norm is defined as the supremum of the set of all
+values of `v (f.coeff R i) * c ^ i` for all `i : ℕ`.
+
+
+## Main Definitions and Results
+* `PowerSeries.gaussNormC` is the supremum of the set of all values of `v (f.coeff R i) * c ^ i`
+  for all `i : ℕ`, where `f` is a power series in `R⟦X⟧`, `v : R → ℝ` is a function and `c` is a
+  real number.
+* `PowerSeries.gaussNormC_nonneg`: if `v` is a non-negative function, then the Gauss norm is
+  non-negative.
+* `PowerSeries.gaussNormC_eq_zero_iff`: if `v` is a non-negative function and `v x = 0 ↔ x = 0` for
+  all `x : R` and `c` is positive, then the Gauss norm is zero if and only if the power series is
+  zero.
+-/
+
+namespace PowerSeries
+
+variable {R F : Type*} [Semiring R] [FunLike F R ℝ] (v : F) (c : ℝ) (f : R⟦X⟧)
+
+/-- Given a power series `f` in `R⟦X⟧`, a function `v : R → ℝ` and a real number `c`, the Gauss norm
+is defined as the supremum of the set of all values of `v (f.coeff R i) * c ^ i` for all `i : ℕ`. -/
+noncomputable def gaussNormC : ℝ := sSup {v (f.coeff R i) * c ^ i | (i : ℕ)}
```

Comment:
Do you think `iSup` is better than `sSup`? i.e. `⨆ i : ℕ, v (f.coeff R i) * c ^ i`